### PR TITLE
Include email request details api into email client

### PR DESCRIPTION
**Pull request by JVRavi**
updated git client to fetch the details about the email request.

### CODE REVIEW

File: src/Kmd.Logic.Email.Client/EmailClient.cs [hunk 1]
```diff
@@ -199,13 +199,38 @@ public async Task<IEnumerable<EmailProviderConfiguration>> GetAllProviderConfigu
...
                 }
             }
         }
 
+        /// <summary>
+        /// Get the information about email request.
+        /// </summary>
+        /// <param name="emailRequestDetails">Email request details.</param>
+        /// <returns>EmailDetailsResponse.</returns>
+        public async Task<EmailDetailsResponse> GetEmailDetails(EmailRequestDetails emailRequestDetails)
```
Comment (smahapatra86) [inline src/Kmd.Logic.Email.Client/EmailClient.cs]: Maybe we can rename it as `GetEmailRequestDetails` to match with the functionality.

File: src/Kmd.Logic.Email.Client/EmailClient.cs [hunk 2]
```diff
@@ -199,13 +199,38 @@ public async Task<IEnumerable<EmailProviderConfiguration>> GetAllProviderConfigu
...
+                switch (emailDetailsResponse?.Response?.StatusCode)
+                {
+                    case System.Net.HttpStatusCode.OK:
+                        return emailDetailsResponse.Body;
+
+                    case System.Net.HttpStatusCode.NotFound:
+                        return null;
+
+                    default:
+                        throw new EmailException(emailDetailsResponse?.Body?.ToString() ?? "Error accessing Email details service.");
```
Comment (smahapatra86) [inline src/Kmd.Logic.Email.Client/EmailClient.cs]: Message can be `Error accessing email service`.

**Review by smahapatra86 — CHANGES_REQUESTED** [review]

File: src/Kmd.Logic.Email.Client/EmailClient.cs (hunk 2 above)
Comment (JVRavi) [reply src/Kmd.Logic.Email.Client/EmailClient.cs]: Changed .

File: src/Kmd.Logic.Email.Client/EmailClient.cs (hunk 1 above)
Comment (JVRavi) [reply src/Kmd.Logic.Email.Client/EmailClient.cs]: Done

**Review by smahapatra86 — APPROVED** [review]

**Review by avinashbr123 — APPROVED** [review]

**Pull request merged**